lib/src/layer/polygon_layer/painter.dart

```diff
@@ -39,42 +39,41 @@ base class _PolygonPainter<R extends Object>
     required LatLng coordinate,
   }) {
     final polygon = projectedPolygon.polygon;
-
-    if (!polygon.boundingBox.contains(coordinate)) return false;
+    if (!polygon.boundingBox.contains(coordinate)) {
+      return false;
+    }
 
     final projectedCoords = getOffsetsXY(
       camera: camera,
       origin: hitTestCameraOrigin,
       points: projectedPolygon.points,
-    ).toList();
+    );
 
     if (projectedCoords.first != projectedCoords.last) {
       projectedCoords.add(projectedCoords.first);
     }
+    final isInPolygon = isPointInPolygon(point, projectedCoords);
 
     final hasHoles = projectedPolygon.holePoints.isNotEmpty;
-    late final List<List<Offset>> projectedHoleCoords;
-    if (hasHoles) {
-      projectedHoleCoords = projectedPolygon.holePoints
-          .map(
-            (points) => getOffsetsXY(
+    final isInHole = hasHoles &&
+        () {
+          for (final points in projectedPolygon.holePoints) {
+            final projectedHoleCoords = getOffsetsXY(
               camera: camera,
               origin: hitTestCameraOrigin,
               points: points,
-            ).toList(),
-          )
-          .toList();
+            );
 
-      if (projectedHoleCoords.firstOrNull != projectedHoleCoords.lastOrNull) {
-        projectedHoleCoords.add(projectedHoleCoords.first);
-      }
-    }
+            if (projectedHoleCoords.first != projectedHoleCoords.last) {
+              projectedHoleCoords.add(projectedHoleCoords.first);
+            }
 
-    final isInPolygon = _isPointInPolygon(point, projectedCoords);
-    final isInHole = hasHoles &&
-        projectedHoleCoords
-            .map((c) => _isPointInPolygon(point, c))
-            .any((e) => e);
```

Comment:
Doing the any so late just means that we're doing more work than needed if if any of the holes other than the last happens to yield true.

Reply:
I'm not sure this is true? `map` outputs a lazy `Iterable`, so I would expect `any` to lazily consume it value by value, and exit early as necessary. But I have no idea. Can we do a small benchmark?

Reply:
Nobody said we can't both be right. You're right but you're still building the entire `projectedHoleCoords` upfront, which may not be needed. This is no longer the caase.